pom.xml

```diff
@@ -241,10 +239,38 @@
                         <artifactId>maven-compiler-plugin</artifactId>
                         <configuration>
                             <showWarnings>true</showWarnings>
+                            <fork>true</fork>
```

Comment:
Hi @jqno, out of curiosity, what made you go with `fork=true` instead of having a `.mvn/jvm.config` configuration?

I'm asking because I'm introducing ErrorProne too, and I picked the `jvm.config` direction so far, so I'm curious about your reasoning 🙂

Reply:
> With a `~/.mvn/jvm.config`, you have to tell people to set that up, and figure something out for running it in CI, and also it will apply to all your other Maven projects on that machine as well. So this just seemed like the easier way 😄

I wasn't precise - I'm configuring these under the Maven Wrapper folder (see spring-projects/spring-batch#4807) so `.mvn/jvm.config` is on git and should be picked up automatically by most users.

(I'm not sure if you run your locally installed Maven and not via `./mnvw`, maybe it still looks for that config? I haven't tried it yet)

Reply:
Ahhh, I didn't think of that possibility. I don't use the Maven Wrapper myself, so I wasn't aware that you can configure it like that. That actually sounds interesting, though I'm not sure I can fix my muscle memory to type `./mvnw` instead of `mvn` :sweat_smile:

Reply:
I just tried to execute my locally installed `mvn`, it seems to work 🙂 and [it's also documented](https://maven.apache.org/configure.html), unrelated to the wrapper:

> Starting with Maven 3.3.1+ you can define JVM configuration via `.mvn/jvm.config` file which means you can define the options for your build on a per project base. This file will become part of your project and will be checked in along with your project. So no need anymore for `MAVEN_OPTS`, `.mavenrc` files.

Reply:
Well then, I guess the answer to your original question is, because I didn't read the documentation :sweat_smile: 

Knowing this, there doesn't seem to be a very good reason for one or the other, apart for personal preference maybe? I think I'll leave it in the pom, at least for now, since it's already working well as it is... Also, I think I like having this stuff close to the rest of the ErrorProne config

Reply:
Oh that's good to know. Might change it then after all :sweat_smile: Thanks!